frontend/plugins/tourism_ui/src/modules/tms/branch-detail/dashboard/amenities/_components/AmenityEditSheet.tsx

```
@@ -49,15 +59,33 @@ export const AmenityEditSheet = ({
+    fieldPaths,
+  } = useAmenityLanguage({ branchLanguages, mainLanguage, fields });
+
   useEffect(() => {
     form.reset({
       name: amenity.name || '',
       icon: amenity.icon || '',
+      translations: buildTranslationsFromAmenity(amenity, translationLanguages),
     });
-  }, [amenity, form]);
+    setSelectedLang(mainLanguage || allLanguages[0] || '');
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)